cou/apps/core.py

```diff
@@ -87,8 +117,7 @@ def _get_unit_upgrade_steps(self, unit: Unit, force: bool) -> UnitUpgradeStep:
         :return: Unit upgrade step
         :rtype: UnitUpgradeStep
         """
-        unit_plan = UnitUpgradeStep(f"Upgrade plan for unit '{unit.name}'")
-        unit_plan.add_step(self._get_disable_scheduler_step(unit))
+        unit_plan = UnitUpgradeStep(description=f"Upgrade plan for unit '{unit.name}'")
```

Comment:
non-blocking: I think it would be better to add a new class `UnitUpgradePlan` to be used here.

Reply:
I agree that that would be nice, it's not related with this PR, so I will suggest to no to it here. We can report an issue about this.

not-blocking: Why you are adding `description=` back? I do not see any benefit of it and in a lot of places I tried to remove it, so make lines shorter.

Reply:
Issue for adding `UnitUpgradePlan` created: https://github.com/canonical/charmed-openstack-upgrader/issues/346